File: 11/row.js
song = [["row", 3], ["your", 1], ["boat", 1]];


function row(song){
  new_arr = [];

  song.forEach(function(each_sub_array){
    for (var i = 0; i < each_sub_array[1]; i++) {
      new_arr.push(each_sub_array[0]);
    }
  });

  return new_arr
}


console.log(row(song));

File: 11/row.rb
song = [["row", 3], ["your", 1], ["boat", 1]]

def row(old)
  new_arr = []
  old.each do |each_sub_array|
    each_sub_array[1].times do
      new_arr << each_sub_array[0]
    end
  end
  return new_arr
end

p row(song)
